### PR TITLE
Qt6 support (pt2)

### PR DESCRIPTION
**Pull request by alexbruy**
Fix QR code scanner.
Use new API and fix deprecations warnings.

### CODE REVIEW

File: app/androidutils.h [hunk 1]
```diff
@@ -10,10 +10,9 @@
 #ifndef ANDROIDUTILS_H
 #define ANDROIDUTILS_H
 
-
 #ifdef ANDROID
-#include <QAndroidActivityResultReceiver>
-#include <QAndroidJniObject>
+#include <QtCore/private/qandroidextras_p.h>
```
Comment (PeterPetrik) [inline app/androidutils.h:14]: uff, why this is needed? I thought androidextras are dead?

File: app/android.pri [hunk 2]
```diff
@@ -74,11 +74,11 @@ android {
     LIBS += -lsqlite3 -liconv -lz -lzip
     LIBS += -ltiff
     LIBS += -lwebp -lcurl -ljpeg
-    
+
     QT += multimedia
     QT += printsupport
-    QT += androidextras
     QT += bluetooth
+    QT += core-private
```
Comment (PeterPetrik) [inline app/android.pri:81]: looks scary? why it is needed?

File: app/androidutils.h (hunk 1 above)
Comment (PeterPetrik) [reply app/androidutils.h:14]: all of those classes should be part of QtCore now no?

Comment (alexbruy) [reply app/androidutils.h:14]: According to https://www.qt.io/blog/qt-extras-modules-in-qt-6 and https://doc-snapshots.qt.io/qt6-6.3/qtandroidprivate.html no, they are not available in the QtCore. There are only few things in the https://doc-snapshots.qt.io/qt6-6.3/qnativeinterface-qandroidapplication.html

File: app/android.pri (hunk 2 above)
Comment (alexbruy) [reply app/android.pri:81]: According to https://www.qt.io/blog/qt-extras-modules-in-qt-6 and https://doc-snapshots.qt.io/qt6-6.3/qtandroidprivate.html we need it as we use lot of the functions which are not yet available in the QtCore

**Pull request merged**